lfsapi/client.go

```diff
@@ -122,10 +124,33 @@ func (c *Client) doWithRedirects(cli *http.Client, req *http.Request, via []*htt
 		return nil, err
 	}
 
-	res, err := cli.Do(req)
+	var retries int
+	if n, ok := Retries(req); ok {
+		retries = n
+	} else {
+		retries = defaultRequestRetries
+	}
+
+	retries = tools.MaxInt(0, retries)
+
+	var res *http.Response
+
+	for i := 0; i < retries; i++ {
+		res, err = cli.Do(req)
+		if err == nil {
+			break
+		}
+
+		if seek, ok := req.Body.(io.Seeker); ok {
+			seek.Seek(0, io.SeekStart)
+		}
+
+		c.traceResponse(req, tracedReq, nil)
+	}
```

Comment:
I think this is where the potentially nil `res` can come from. If it fails retries enough times, `res` is never set. 

Is this the right place to put the retry logic though? How would it interfere with the transfer adapter retries?

Reply:
> Is this the right place to put the retry logic though? How would it interfere with the transfer adapter retries?

Good question -- there shouldn't be any interference as-is, since clients of the `lfsapi` package have to explicitly opt-in for more than 1 request per `Do()` call. Since `tq` manages its own retries, omitting `WithRetries()` is sufficient to keep behavior the same.